plugins/UM3NetworkPrinting/PrintJobInfoBlock.qml

```diff
@@ -0,0 +1,378 @@
+                text: printJob.name
+                font: UM.Theme.getFont("default_bold")
+                width: parent.width
+                elide: Text.ElideRight
+            }
+
+            Label
+            {
+                id: ownerName
+                anchors.top: printJobName.bottom
+                text: printJob.owner
+                font: UM.Theme.getFont("default")
+                opacity: 0.6
+                width: parent.width
+                elide: Text.ElideRight
+            }
+
+            Image
+            {
+                id: printJobPreview
+                source: printJob.previewImageUrl
+                anchors.top: ownerName.bottom
+                anchors.horizontalCenter: parent.horizontalCenter
+                anchors.bottom: totalTimeLabel.bottom
+                width: height
+                opacity: printJob.state == "error" ? 0.5 : 1.0
+            }
+
+            UM.RecolorImage
+            {
+                id: statusImage
+                anchors.centerIn: printJobPreview
+                source: printJob.state == "error" ? "aborted-icon.svg" : ""
+                visible: source != ""
+                width: 0.5 * printJobPreview.width
+                height: 0.5 * printJobPreview.height
+                sourceSize.width: width
+                sourceSize.height: height
+                color: "black"
+            }
+
+            Label
+            {
+                id: totalTimeLabel
+                opacity: 0.6
+                anchors.bottom: parent.bottom
+                anchors.right: parent.right
+                font: UM.Theme.getFont("default")
+                text: printJob != null ? getPrettyTime(printJob.timeTotal) : ""
+                elide: Text.ElideRight
+            }
+        }
+
+        Item
+        {
+            // Content on the right side of the infobox.
+            anchors
+            {
+                top: parent.top
+                bottom: parent.bottom
+                left: parent.horizontalCenter
+                right: parent.right
+                margins: 2 * UM.Theme.getSize("default_margin").width
+                leftMargin: UM.Theme.getSize("default_margin").width
+            }
+
+            Label
+            {
+                id: targetPrinterLabel
+                elide: Text.ElideRight
+                font: UM.Theme.getFont("default_bold")
+                text:
+                {
+                    if(printJob.assignedPrinter == null)
+                    {
+                        if(printJob.state == "error")
+                        {
+                            return catalog.i18nc("@label", "Waiting for: Unavailable printer")
+                        }
+                        return catalog.i18nc("@label", "Waiting for: First available")
+                    }
+                    else
+                    {
+                        return catalog.i18nc("@label", "Waiting for: ") + printJob.assignedPrinter.name
```

Comment:
Ideally this would use formatting for the i18n string and not assume so much about how every language should build up this bit of text. Also it really should be combined with the i18n above. The "Waiting for: " bit is meant to be the same although the translators might not realise it.

Reply:
I see what you're saying but I'm not experienced enough with i18nc to know what to do about it.